statistics/handle/handle.go

```diff
@@ -121,6 +122,9 @@ func (h *Handle) withRestrictedSQLExecutor(ctx context.Context, fn func(context.
 
 func (h *Handle) execRestrictedSQL(ctx context.Context, sql string, params ...interface{}) ([]chunk.Row, []*ast.ResultField, error) {
 	return h.withRestrictedSQLExecutor(ctx, func(ctx context.Context, exec sqlexec.RestrictedSQLExecutor) ([]chunk.Row, []*ast.ResultField, error) {
+		if variable.TopSQLEnabled() {
+			defer pprof.SetGoroutineLabels(ctx)
```

Comment:
Maybe better to add a comment that this is to restore the label by using the original ctx after execution is finished.

Reply:
done;